setup_no_spack.sh

```diff
@@ -1,41 +1,85 @@
-# !/usr/bin/env bash
+#!/usr/bin/env bash
 
 test_folder="tiny_CI_test"
 
-export PATH="/opt/homebrew/bin:$PATH"
-export LDFLAGS="-L/opt/homebrew/opt/openmpi/lib"
-export CPPFLAGS="-I/opt/homebrew/opt/openmpi/include"
+if [[ "$(uname -s)" == "Darwin" ]]; then
+    PLATFORM=mac
+elif [[ -f /sys/class/dmi/id/sys_vendor ]] && grep -qi 'microsoft' /sys/class/dmi/id/sys_vendor; then
+    PLATFORM=azure
+elif [[ -f /etc/os-release ]] && grep -qi '^name="Amazon Linux"' /etc/os-release; then
+    PLATFORM=aws
+else
```

Comment:
Can we also have another vanilla "Linux" option as well? I attempted to install to my local ubuntu VM and it stops since it is neither azure nor aws

Reply:
maybe it can go with azure. I did not have something to test it. 

Issue here: #43

yes, this was not the aim of this pr. I would move forward and track this in another pr